### PR TITLE
[network/conntrack/ebpf] don't fail loading system-probe if nf_conntrack kernel module is not loaded

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

check if a nf_conntrack symbol is present/loaded in the kernel and return a noOpConntrack if not 

### Motivation

Avoid system-probe loading issue due to a dependency

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/ebpf_conntracker.go [hunk 1]
```diff
@@ -57,6 +59,13 @@ type ebpfConntracker struct {
 
 // NewEBPFConntracker creates a netlink.Conntracker that monitor conntrack NAT entries via eBPF
 func NewEBPFConntracker(cfg *config.Config) (netlink.Conntracker, error) {
+	ksymPath := filepath.Join(cfg.ProcRoot, "kallsyms")
+	missing, err := ddebpf.VerifyKernelFuncs(ksymPath, []string{"__nf_conntrack_hash_insert"})
+	if err == nil && len(missing) > 0 {
+		log.Warnf("could not initialize ebpf conntrack, tracer will continue without NAT tracking: nf_conntrack kernel module is not loaded")
```
[the remark this answers is not in the document]
Comment (nplanel) [reply pkg/network/tracer/ebpf_conntracker.go]: updated the comment as we discuss it on slack

File: pkg/network/tracer/ebpf_conntracker.go [hunk 2]
```diff
@@ -57,6 +59,13 @@ type ebpfConntracker struct {
 
 // NewEBPFConntracker creates a netlink.Conntracker that monitor conntrack NAT entries via eBPF
 func NewEBPFConntracker(cfg *config.Config) (netlink.Conntracker, error) {
+	ksymPath := filepath.Join(cfg.ProcRoot, "kallsyms")
+	missing, err := ddebpf.VerifyKernelFuncs(ksymPath, []string{"__nf_conntrack_hash_insert"})
```
Comment (brycekahle) [inline pkg/network/tracer/ebpf_conntracker.go:63]: Does the netlink-based conntracker work if the kernel module is not loaded? Perhaps this should be a level higher and apply to both netlink and eBPF conntrackers?

File: pkg/network/tracer/ebpf_conntracker.go [hunk 3]
```diff
@@ -57,6 +58,13 @@ type ebpfConntracker struct {
 
 // NewEBPFConntracker creates a netlink.Conntracker that monitor conntrack NAT entries via eBPF
 func NewEBPFConntracker(cfg *config.Config) (netlink.Conntracker, error) {
+	// dial the netlink layer aim to load nf_conntrack_netlink and nf_conntrack kernel modules
+	// eBPF conntrack require nf_conntrack symbols
+	conn, err := libnetlink.Dial(unix.NETLINK_NETFILTER, nil)
```
Comment (brycekahle) [inline pkg/network/tracer/ebpf_conntracker.go:63]: If this fails, do we need to return an error?

Comment (nplanel) [reply pkg/network/tracer/ebpf_conntracker.go:63]: We will not be able to load ebpf or netlink conntrack so we will go the same error as today. The corner scenario would be if you upgrade your kernel but didn't reboot (the modules can't be loaded anymore)

**Review by brycekahle — APPROVED** [review]

**Pull request merged**